src/statisticalmodel.jl

```diff
@@ -314,3 +314,13 @@ function adjr2(model::StatisticalModel, variant::Symbol)
 end
 
 const adjr² = adjr2
+
+"""
+    momentmatrix(model::StatisticalModel)
```

Comment:
Why not define this for `RegressionModel` instead? It seems more similar to `modelmatrix` and it only makes sense for regressions, right?

Reply:
Not really. The idea is to use moment matrix for statistical model that are defined through moment conditions like General Estimating Equations and Generalized Method of Moments.